http_service/tests/test_schedule_tests.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from bugbug_http.models import schedule_tests
+
+
+def test_simple_schedule(patch_resources, mock_hgmo, mock_repo):
+
+    # The repo should be almost empty at first
+    repo_dir, repo = mock_repo
+    assert len(repo.log()) == 4
+    test_txt = repo_dir / "test.txt"
+    assert test_txt.exists()
+    assert test_txt.read_text("utf-8") == "Version 3"
+
+    # Scheduling a test on a revision should apply changes in the repo
+    assert schedule_tests("mozilla-central", "12345deadbeef") == "OK"
+
+    # Check changes have been applied
+    assert len(repo.log()) == 5
+    assert test_txt.read_text("utf-8") == "Version 3\nThis is a new line\n"
+
+
+@pytest.mark.parametrize(
+    "branch, revision, result, final_log",
+    [
+        # patch from autoland based on local parent n°0
+        (
+            "integration/autoland",
+            "normal123",
+            "OK",
+            ["Target patch", "Parent 123", "Base history 0"],
+        ),
+        # patch from autoland where parent is not available
+        # so the patch is rejected as we need the parents on autoland
+        # and no changes is made on the repository
+        (
+            "integration/autoland",
+            "orphan123",
+            "NOK",
+            ["Base history 3", "Base history 2", "Base history 1", "Base history 0"],
+        ),
+        # patch from try based on local parent n°1
+        (
+            "try",
+            "normal456",
+            "OK",
+            ["Target patch", "Parent 456", "Base history 1", "Base history 0"],
+        ),
+        # patch from try where parent is not available
+        # so the patch is applied on top of tip
+        (
+            "try",
+            "orphan456",
+            "OK",
+            [
+                "Orphan 456",
+                "Base history 3",
+                "Base history 2",
+                "Base history 1",
+                "Base history 0",
+            ],
+        ),
+    ],
+)
+def test_schedule(
```

Comment:
It'd be nice to also have a test where the parent is not available in the local repo but is available on the remote repo and so we can update to it after pulling

Reply:
(where the patch wouldn't apply without the parent, but would apply with the parent)

Reply:
Can be done in a follow-up (as part of #1371).